fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/AuthenticatedRequestAuthenticationConverter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Apple, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.shared;
+
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.web.authentication.AuthenticationConverter;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+
+public class AuthenticatedRequestAuthenticationConverter implements AuthenticationConverter {
```

Comment:
sorry to keep nitpicking...I need all the help I can get keeping the various bits and pieces here straight.  javadoc please.

Reply:
I documented that on the bean that constructs it. Or do you want it in both places? Moved? I'm not sure where the most appropriate place would be to document those as the settings are sort of mutually incompatible being the same interface for a singleton bean and all.

Reply:
I definitely like the comment on the bean method.  Maybe the same comment here makes sense as well (without "When enabled..")?  I would like to see something here that explains what the class is for.

Reply:
Thank you!